src/com/pentalog/pentastagiu/homework3/ex2/Basket.java

```diff
@@ -0,0 +1,51 @@
+package com.pentalog.pentastagiu.homework3.ex2;
+
+public class Basket {
+    private double totalPrice;
+    private Costumer costumer;
+
+    public Basket(Costumer costumer) {
+        this.costumer = costumer;
+    }
+
+    private void addProduct(Product produs) {
+        this.totalPrice += produs.getPrice();
+        System.out.println("The product was successfully added!! Total price = " + this.totalPrice + " lei.");
+    }
+
+    private void infoBasket() {
+        switch (this.costumer.getStatus()) {
+            case Gold:
+                System.out.println("Total price: " + this.totalPrice + " ; Reduced price: "
+                        + (this.totalPrice - 0.2 * this.totalPrice));
```

Comment:
e ok, dar si mai bine ar fi ca valoarea discountului sa o pastrezi in enum, sa ai acolo un field private discount si un getter public pentru ca ai avea asa un mod centraliat de pastrare a acestei relatii intre statusul unui csutomer si disountul aplicat, de care te-ai putea refolosi apoi si in alte locuri din aplicatie; daca apoi vei vrea sa faci schimbari(customer GOLD sa aiba discount de 25% in loc de 20%) vei face modificarea asta doar in enum si nu va fi nevoie sa intervii in alte bucati de cod care se folosesc de aceasta relatie